project/ValidatePullRequest.scala

```diff
@@ -3,19 +3,137 @@
+  val SourceBranchJenkinsEnvVarName = "ghprbSourceBranch"
+  val targetBranch = settingKey[String]("Branch with which the PR changes should be diffed against")
+
+  // determining touched dirs and projects
+  val changedDirectories = taskKey[immutable.Set[String]]("List of touched modules in this PR branch")
+  val projectIsAffectedByChanges = taskKey[Boolean]("True if this project is affected by the PR and should be rebuilt")
+
+  // running validation
   val validatePullRequest = taskKey[Unit]("Additional tasks for pull request validation")
 
   override def trigger = allRequirements
 
   override lazy val projectSettings = Seq(
-    validatePullRequest := (),
-    validatePullRequest <<= validatePullRequest.dependsOn(test in Test),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.PerformanceTest"),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.LongRunningTest"),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.TimingTest"),
+
+    targetBranch in ValidatePR := {
+      List(
+        System.getProperty(TargetBranchEnvVarName),
+        System.getProperty(TargetBranchJenkinsEnvVarName), // Set by "GitHub pull request builder plugin"
+        "master").find(_ != null).get
+    },
+
+    sourceBranch in ValidatePR := {
+      List(
+        System.getProperty(SourceBranchEnvVarName),
+        System.getProperty(SourceBranchJenkinsEnvVarName), // Set by "GitHub pull request builder plugin"
+        "HEAD").find(_ != null).get
+    },
+
+    changedDirectories in ValidatePR := {
+      val log = streams.value.log
+
+      val targetId = (targetBranch in ValidatePR).value
+      val prId = (sourceBranch in ValidatePR).value
+
+      // TODO could use jgit
+      log.info(s"Comparing [$targetId] with [$prId] to determine changed modules in PR...")
+      val gitOutput = "git diff %s..%s --stat".format(targetId, prId).!!.split("\n")
+
+      val moduleNames =
+        gitOutput
+          .filterNot(_ contains " files changed, ")
+          .map(l ⇒ l.trim.takeWhile(_ != '/'))
+          .filterNot(_ == "project")
+          .toSet
+
+      log.info("Detected changes in directories: " + moduleNames.mkString("[", ", ", "]"))
+      moduleNames
+    },
+
+    projectIsAffectedByChanges in ValidatePR := {
+      val log = streams.value.log
+      log.debug("Analysing project (for inclusion in PR validation): " + name.value)
+
+      // if in any scope, any of the changed modules is within this projects dependencies, we must test it:
+      val shouldBeBuilt = (changedDirectories in ValidatePR).value.exists { modifiedProject ⇒
+        Set(Compile, Test, Runtime, Provided, Optional) exists { ivyScope: sbt.Configuration ⇒
+          log.debug(s"Analysing $ivyScope scoped dependencies...")
+          val modifiedModule = modifiedProject match {
+            case p if p contains "persistence" => ModuleId("com.typesafe.akka", s"$p-experimental_2.11", "2.4-SNAPSHOT")
```

Comment:
Fixed now, no need for the match! (if any "experimental" or "non experimental" appended names happen to be in a dependency we want to test it)